contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -37,18 +38,26 @@ def prepare(cls, options, round_manager):
     super(MypyTask, cls).prepare(options, round_manager)
     round_manager.require_data(PythonInterpreter)
 
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(MypyTask, cls).subsystem_dependencies() + (PythonRepos, PythonSetup)
+
   @classmethod
   def register_options(cls, register):
-    register('--mypy-version', default='0.550', help='The version of mypy to use.')
-    register('--config-file', default=None,
+    super(MypyTask, cls).register_options(register)
+    register('--mypy-version', default='0.580', fingerprint=True,
+             help='The version of mypy to use.')
+    register('--config-file', default=None, fingerprint=True,
```

Comment:
Yes, this should be fingerprinted. To make sure it gets deeply fingerprinted, should pass `type=file_option`.